x-pack/plugins/rollup/__jest__/integration_tests/job_create.test.js

```diff
@@ -0,0 +1,1298 @@
+    };
+    server.respondWith(/\/api\/rollup\/index_pattern_validity\/.*/, [
+      200,
+      { 'Content-Type': 'application/json' },
+      JSON.stringify({ ...defaultResponse, ...response }),
+    ]);
+  };
+
+  const mockCreateJob = () => {
+    server.respondWith(/\/api\/rollup\/create/, [
+      200,
+      { 'Content-Type': 'application/json' },
+      JSON.stringify({}),
+    ]);
+  };
+
+  mockIndexPatternValidityResponse();
+  mockCreateJob();
+
+  return { mockIndexPatternValidityResponse };
+};
+
+describe('Create Rollup Job', () => {
+  let server;
+  let findTestSubject;
+  let testSubjectExists;
+  let userActions;
+  let getFormErrorsMessages;
+  let getMetadataFromEuiTable;
+  let form;
+  let mockIndexPatternValidityResponse;
+  let getEuiStepsHorizontalActive;
+  let goToStep;
+
+  beforeEach(() => {
+    server = sinon.fakeServer.create();
+    server.respondImmediately = true;
+    ({ mockIndexPatternValidityResponse } = mockServerResponses(server));
+    ({
+      findTestSubject,
+      testSubjectExists,
+      userActions,
+      getFormErrorsMessages,
+      getMetadataFromEuiTable,
+      form,
+      goToStep,
+      getEuiStepsHorizontalActive,
+    } = initTestBed());
+  });
+
+  afterEach(() => {
+    server.restore();
+  });
+
+  describe('Step 1: Logistics', () => {
+    it('should have the horizontal step active on "Logistics"', () => {
+      expect(getEuiStepsHorizontalActive()).toContain('Logistics');
+    });
+
+    it('should have the title set to "Logistics"', () => {
+      expect(testSubjectExists('rollupJobCreateLogisticsTitle')).toBe(true);
+    });
+
+    it('should have a link to the documentation', () => {
+      expect(testSubjectExists('rollupJobCreateLogisticsDocsButton')).toBe(true);
+    });
+
+    it('should only have the "next" button visible', () => {
+      expect(testSubjectExists('rollupJobBackButton')).toBe(false);
+      expect(testSubjectExists('rollupJobNextButton')).toBe(true);
+      expect(testSubjectExists('rollupJobSaveButton')).toBe(false);
+    });
+
+    it('should display errors when clicking "next" without filling the form', () => {
+      expect(testSubjectExists('rollupJobCreateStepError')).toBeFalsy();
+
+      userActions.clickNextStep();
+
+      expect(testSubjectExists('rollupJobCreateStepError')).toBeTruthy();
+      expect(getFormErrorsMessages()).toEqual([
+        'Name is required.',
+        'Index pattern is required.',
+        'Rollup index is required.',
+      ]);
+      expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(true);
+    });
+
+    describe('form validations', () => {
+      describe('index pattern', () => {
+        beforeEach(() => {
+          expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(false);
+        });
+
+        afterEach(() => {
+          expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(true);
+        });
+
+        it('should not allow spaces', async () => {
```

Comment:
Many of these tests seem more appropriate as unit tests specific to the validation functions in `steps_config`, though maybe I'm just not understanding the testing paradigm we're using here. Can you help me understand why they should be here instead?

Reply:
So the idea is to be as close as possible to how our app (or app _section_ in this case) will be used by a user. If we test in isolation all the files in `step_configs`:

- We'll create a lot of files 😊 
- We don't have any guarantee that our component uses them. So we might have a green CI but a broken app.

Here we have the best of both world (functional and unit testing) without the complexity of a full browser + API integration testing. It also allows us to refactor without worying about the implementation details.

And... if a product/QA person reads the spec files, he understands a first sight and in 1 place the validation of our form, withouth the need to go throught dozens of files and hoping things are wired correctly.